src/main/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStepExecution.java

```diff
@@ -229,4 +225,63 @@ private void purgeSockets() {
         return socket;
     }
 
+    /**
+     * Retrieves the workspace reference in the current context.
+     *
+     * @return File path object for the workspace
+     *
+     * @throws IOException See {@link StepContext#get(Class)}
+     * @throws InterruptedException See {@link StepContext#get(Class)}
+     */
+    private FilePath getWorkspace() throws IOException, InterruptedException {
+        return getContext().get(FilePath.class);
+    }
+
+    /**
+     * Retrieves the launcher reference in the current context.
+     *
+     * @return Current launcher object
+     *
+     * @throws IOException See {@link StepContext#get(Class)}
+     * @throws InterruptedException See {@link StepContext#get(Class)}
+     */
+    private Launcher getLauncher() throws IOException, InterruptedException {
+        return getContext().get(Launcher.class);
+    }
+
+    /**
+     * Retrieves the build reference in the current context.
+     *
+     * @return Current build/run object
+     *
+     * @throws IOException See {@link StepContext#get(Class)}
+     * @throws InterruptedException See {@link StepContext#get(Class)}
+     */
+    private Run<?, ?> getBuild() throws IOException, InterruptedException {
+        return getContext().get(Run.class);
+    }
+
+    /**
+     * Retrieves the task listener reference in the current context.
+     *
+     * @return Current task listener object
+     *
+     * @throws IOException See {@link StepContext#get(Class)}
+     * @throws InterruptedException See {@link StepContext#get(Class)}
+     */
+    private TaskListener getListener() throws IOException, InterruptedException {
```

Comment:
Tip: you can avoid `catch`ing these at call sites by falling back to either `TaskListener.NULL` or `LogTaskListener`.

Reply:
Did not know of that, changed.